core/src/main/java/com/yahoo/oak/BasicChunk.java

```diff
@@ -9,55 +9,503 @@
+        // The allocated slice includes all the needed information for further access,
+        // the reference is set in the slice as part of the alocation
+        ctx.newValue.getSlice().allocate(valueDataSize, writeForMove);
+        ctx.isNewValueForMove = writeForMove;
+
+        ScopedWriteBuffer.serialize(ctx.newValue.getSlice(), value, valueSerializer);
+    }
+
+    /**
+     * writeValueCommit does the physical CAS of the value reference, which is the Linearization
+     * Point of the insertion.
+     *
+     * @param ctx The context that follows the operation since the key was found/created.
+     *            Holds the entry index to which the value reference is linked, the old and new
+     *            value references.
+     *
+     * @return TRUE if the value reference was CASed successfully.
+     */
+    ValueUtils.ValueResult writeValueCommit(ThreadContext ctx) {
+        // If the commit is for a writing the new value, the old values should be invalid.
+        // Otherwise (commit is for moving the value) old value reference is saved in the context.
+
+        long oldValueReference = ctx.value.getSlice().getReference();
+        long newValueReference = ctx.newValue.getSlice().getReference();
+        assert valuesMemoryManager.isReferenceValid(newValueReference);
+
+        if (!casValueReference(ctx.entryIndex, oldValueReference, newValueReference)) {
+            return ValueUtils.ValueResult.FALSE;
+        }
+        return ValueUtils.ValueResult.TRUE;
+    }
+
+    /**
+     * Releases the key of the input context.
+     * Currently in use only for unreached keys, waiting for GC to be arranged
+     *
+     * @param ctx the context that follows the operation since the key was found/created
+     **/
+    void releaseKey(ThreadContext ctx) {
+        // Keys are now managed via Sequentially Expanding Memory Manager, but since this key's slice
+        // can not be reached or used by other thread it is OK to release it and to allocate again.
+        ctx.key.getSlice().release();
+    }
+
+    /**
+     * Releases the newly allocated value of the input context.
+     * Currently the method is used only to release an
+     * unreachable value reference, the one that was not yet attached to an entry!
+     * The method is part of EntryArray, because it cares also
+     * for writing the value before attaching it to an entry (allocateValue/writeValueCommit)
+     *
+     * @param ctx the context that follows the operation since the key was found/created
+     **/
+    void releaseNewValue(ThreadContext ctx) {
+        ctx.newValue.getSlice().release();
+    }
+
+    /**
+     * Completes the deletion of a mapping
+     * @param ctx the call context
+     * @return true if the deletion indeed updated the entry to be deleted as a unique operation
+     */
+    abstract boolean deleteValueFinish(ThreadContext ctx);
+
+
+    /* ########################################################################
+       # Former Chunk
```

Comment:
What does it mean "Former Chunk"?

Reply:
This is a comment for the reviewer and later will be removed. Before this PR, the methods below it were implemented in Chunk, and the methods above it were implemented in EntryArray.

Reply:
OK